### PR TITLE
Add documentation on how to use `USING`

### PR DESCRIPTION
**Pull request by talgalili**
(4th attempt at this point, combining github online/desktop and terminal did not go well for me...)

### CODE REVIEW

File: presto-docs/src/main/sphinx/sql/select.rst [hunk 1]
```diff
@@ -804,6 +804,82 @@ The following query will fail with the error ``Column 'name' is ambiguous``::
...
+USING
+^^^^^
+The ``USING`` clause allows you to write shorter queries when both tables you 
+are joining have the same name for the join key.
+
+For example:
+
+.. code-block:: none 
+
+    SELECT *
```
Comment (highker) [inline presto-docs/src/main/sphinx/sql/select.rst]: Could you use tpch tables as examples like other examples do?

File: presto-docs/src/main/sphinx/sql/select.rst [hunk 2]
```diff
@@ -804,6 +804,82 @@ The following query will fail with the error ``Column 'name' is ambiguous``::
...
 
+
+USING
+^^^^^
+The ``USING`` clause allows you to write shorter queries when both tables you 
+are joining have the same name for the join key.
+
+For example:
+
+.. code-block:: none
```
Comment (highker) [inline presto-docs/src/main/sphinx/sql/select.rst]: This is actually a sql code block. Use `::` after "For example"

File: presto-docs/src/main/sphinx/sql/select.rst [hunk 3]
```diff
@@ -804,6 +804,82 @@ The following query will fail with the error ``Column 'name' is ambiguous``::
...
+For example:
+
+.. code-block:: none 
+
+    SELECT *
+    FROM table_1 
+    JOIN table_2
+    ON table_1.key_A = table_2.key_A AND table_1.key_B = table_2.key_B
+
+Can be rewritten to:
```
Comment (highker) [inline presto-docs/src/main/sphinx/sql/select.rst]: "Can" -> "can"

File: presto-docs/src/main/sphinx/sql/select.rst [hunk 4]
```diff
@@ -804,6 +804,82 @@ The following query will fail with the error ``Column 'name' is ambiguous``::
...
+
+    SELECT *
+    FROM table_1
+    JOIN table_2
+    USING (key_A, key_B)
+
+
+The output of doing ``JOIN`` with ``USING`` will be one copy of the join key 
+columns (key_1 and key_2 in the example above) followed by the remaining columns
+in table_1 and then the remaining columns in table_2. Note that the join keys are not
```
Comment (highker) [inline presto-docs/src/main/sphinx/sql/select.rst]: use quotes around example names (e.g., \`\`table_1\`\` or \`\`key_1\`\`)

File: presto-docs/src/main/sphinx/sql/select.rst [hunk 5]
```diff
@@ -804,6 +804,82 @@ The following query will fail with the error ``Column 'name' is ambiguous``::
...
+.. code-block:: none 
+
+    SELECT *
+    FROM table_1 
+    JOIN table_2
+    ON table_1.key_A = table_2.key_A AND table_1.key_B = table_2.key_B
+
+Can be rewritten to:
+
+.. code-block:: none
```
Comment (highker) [inline presto-docs/src/main/sphinx/sql/select.rst]: Same ::

File: presto-docs/src/main/sphinx/sql/select.rst [hunk 6]
```diff
@@ -804,6 +804,82 @@ The following query will fail with the error ``Column 'name' is ambiguous``::
...
+The output of doing ``JOIN`` with ``USING`` will be one copy of the join key 
+columns (key_1 and key_2 in the example above) followed by the remaining columns
+in table_1 and then the remaining columns in table_2. Note that the join keys are not
+included in the list of columns from the origin tables for the purpose of
+referencing them in the query. You cannot access them with a table prefix and 
+if you run ``SELECT table_1.*, table_2.*``, the join columns are not included in the output.
+
+The following two queries are equivalent:
+
+.. code-block:: none
```
Comment (highker) [inline presto-docs/src/main/sphinx/sql/select.rst]: Same ::

File: presto-docs/src/main/sphinx/sql/select.rst (hunk 1 above)
Comment (talgalili) [reply presto-docs/src/main/sphinx/sql/select.rst]: I'm not sure what you mean here with "tpch tables". Can you help direct me to what it is you are referring to? (As far as I can see, other places in the doc uses VALUES for examples that can be ran, and the pseudo code I'm using at the beginning of the section, I feel, is better than a real example)

**Comment by talgalili** [comment]
Hi @highker,
Thanks for the comments!
I've implemented all of your suggestions (except one, which I didn't fully understand - please see my comment there).
I've squashed all the commits into one commit.
If this is good with you, feel free to pull/rebase/merge. If you want me to do more on the diff, please let me know.

Cheers,
T

File: presto-docs/src/main/sphinx/sql/select.rst (hunk 1 above)
Comment (highker) [reply presto-docs/src/main/sphinx/sql/select.rst]: You could start `TpchQueryRunner` and use the tables in side that catalog.
After that you can query against tpch tables
```
cd presto/presto-cli/target
presto-cli-*-executable.jar --catalog tpch --schema sf10
```

All the examples we use are from tpch; so users can directly copy and paste a query on that webpage and run it from cli.

Comment (talgalili) [reply presto-docs/src/main/sphinx/sql/select.rst]: Got it.
Thanks for the suggestion, but I suspect this would only make the example more cumbersome. 
The first examples I wrote are not to be run at the terminal, but this trick has been used throughout the text (e.g.: in the "WITH Clause" section), and is (IMHO) a better way to explain the structure.

The last code I added is self contained (and based on VALUES) and the user can run it in the terminal to see how it behaves (this was also used in the text, e.g.: the sections with union, intersect, UNNEST, etc.).

In short, I don't think adding another example would be helpful to the text (and I think this diff is probably good to go as is).

If you have a specific example you think would be helpful, please share it so that I'd better understand what you had in mind.

What do you think?

File: presto-docs/src/main/sphinx/sql/select.rst [hunk 7]
```diff
@@ -804,6 +804,82 @@ The following query will fail with the error ``Column 'name' is ambiguous``::
...
+The output of doing ``JOIN`` with ``USING`` will be one copy of the join key 
+columns (``key_1`` and ``key_2`` in the example above) followed by the remaining columns
+in ``table_1`` and then the remaining columns in ``table_2``. Note that the join keys are not
+included in the list of columns from the origin tables for the purpose of
+referencing them in the query. You cannot access them with a table prefix and 
+if you run ``SELECT table_1.*, table_2.*``, the join columns are not included in the output.
+
+The following two queries are equivalent::
+
+.. code-block:: none
```
Comment (highker) [inline presto-docs/src/main/sphinx/sql/select.rst]: remove

File: presto-docs/src/main/sphinx/sql/select.rst [hunk 8]
```diff
@@ -804,6 +804,82 @@ The following query will fail with the error ``Column 'name' is ambiguous``::
...
+.. code-block:: none 
+
+    SELECT *
+    FROM table_1 
+    JOIN table_2
+    ON table_1.key_A = table_2.key_A AND table_1.key_B = table_2.key_B
+
+can be rewritten to::
+
+.. code-block:: none
```
Comment (highker) [inline presto-docs/src/main/sphinx/sql/select.rst]: remove

File: presto-docs/src/main/sphinx/sql/select.rst [hunk 9]
```diff
@@ -804,6 +804,82 @@ The following query will fail with the error ``Column 'name' is ambiguous``::
...
 
+
+USING
+^^^^^
+The ``USING`` clause allows you to write shorter queries when both tables you 
+are joining have the same name for the join key.
+
+For example::
+
+.. code-block:: none
```
Comment (highker) [inline presto-docs/src/main/sphinx/sql/select.rst]: remove

File: presto-docs/src/main/sphinx/sql/select.rst [hunk 10]
```diff
@@ -804,6 +804,82 @@ The following query will fail with the error ``Column 'name' is ambiguous``::
...
+.. code-block:: none 
+
+    SELECT *
+    FROM table_1
+    JOIN table_2
+    USING (key_A, key_B)
+
+
+The output of doing ``JOIN`` with ``USING`` will be one copy of the join key 
+columns (``key_1`` and ``key_2`` in the example above) followed by the remaining columns
```
Comment (highker) [inline presto-docs/src/main/sphinx/sql/select.rst]: typo: `key_A`/`key_B`

Comment (talgalili) [reply presto-docs/src/main/sphinx/sql/select.rst]: Good catch, thanks :)

**Comment by talgalili** [comment]
@highker - thanks for the comments, I've implemented all of them, committed, and squashed. I believe this diff is now ready to be rebased/merged.

Cheers,
T

**Comment by highker** [comment]
Thanks @gal for your contribution!
